rs/replicated_state/src/replicated_state.rs

```diff
@@ -707,14 +719,17 @@ impl ReplicatedState {
         subnet_available_memory: &mut i64,
     ) -> Result<(), (StateError, RequestOrResponse)> {
         let own_subnet_type = self.metadata.own_subnet_type;
-        let input_queue_type = if msg.sender().get_ref() == self.metadata.own_subnet_id.get_ref()
-            || self.canister_states.contains_key(&msg.sender())
+        let sender = msg.sender();
+        let input_queue_type = if sender.get_ref() == self.metadata.own_subnet_id.get_ref()
+            || self.canister_states.contains_key(&sender)
         {
             InputQueueType::LocalSubnet
         } else {
             InputQueueType::RemoteSubnet
         };
-        match self.canister_state_mut(&msg.receiver()) {
+
+        let receiver = msg.receiver();
```

Comment:
Is there a particular reason for explicitly assigning `receiver` and `sender` above, or is it simply personal preference that motivated this change? [No opinion at all, just curious whether I'm missing something]

Reply:
It's just that I noticed that both `sender()` and `receiver()` clone the respective `CanisterIds`. And each of them is potentially used multiple times. So it's just a meaningless optimization.

Reply:
I see. Thank you.